api/v1beta1/hostedcluster_types.go

```diff
@@ -434,6 +434,7 @@ type HostedClusterSpec struct {
 
 // OLMCatalogPlacement is an enum specifying the placement of OLM catalog components.
 // +kubebuilder:validation:Enum=management;guest
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
 type OLMCatalogPlacement string
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
// +immutable
type OLMCatalogPlacement string
```
I think we need `// +immutable` for the docs.